src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const listArrays = sourceString
+    .split(';')
+    .filter((item) => item.length)
+    .map((element) => element.split(':').map((item) => item.trim()));
+
+  const resultObject = {};
+  const filterArray = listArrays.reduce((acc, value) => {
+    const [key, val] = value;
+
+    acc[key] = val;
+
+    return acc;
```

Comment:
Using `reduce` here is a good approach, but you don't need to pass `resultObject` as the initial value for the accumulator. Instead, you can use an empty object `{}` directly in the `reduce` call. This will avoid any potential side effects of mutating the `resultObject` outside of the `reduce` scope.

Reply:
ok